src/excmds.ts

```diff
@@ -1542,6 +1562,30 @@ export async function qall() {
 
 // }}}
 
+// {{{ CONTAINERS
+
+/** Closes all tabs open in the same container across all windows.
+ * @param containerId
+ *      The string represtation of the container id.
+ */
+//#background
+export async function containerclose(containerId: string) {
```

Comment:
What are these IDs? Are they just the container names? If they aren't, we should accept names as well as IDs and convert the names into IDs.

We should also have an excmd that lists all of your containers.

Reply:
The container ids are cookieStoreIds from firefox' contextual identities api.  They look like this: `firefox-container-1` and just increment that last digit by 1. 
I went with close by id digit instead of name because of how other addons handle containers. Most allow multiple containers with the same name, color and icon, the constraint placed in this is on (name, color, icon) meaning that even in this implementation you can create containers with the same name.

Reply:
I don't think I care about protecting people from themselves. I think

`tabopen -c blah` should open the container `blahrgbizness`, but throw an error if there is also `blahb` or indeed another `blahrgbizness`.

Related: how do containers get colours with the Firefox addon? is that something built in? It would be nice to add to the mode indicator, like we do for private mode now.

Reply:
So, essentially a fuzzy match on the container name and error it in case of ambiguity? I'll see what I can do.

Re: colours
The colours are built in and easily accessible. I'll take a stab at adding it to the mode indicator.

Reply:
> So, essentially a fuzzy match on the container name and error it in case of ambiguity? I'll see what I can do.

Something like that. I'd probably check for exact matches, then fall back to `containername.includes(requestedname)`.

---

Colours - great :)

Don't sweat it too much if you can't but it should be pretty easy to copy the code that changes it for private.